newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/VerticeRow.js

```diff
@@ -1,66 +1,134 @@
+import { dropIndicatorColor } from '../../../../UI/SortableVirtualizedItemList/DropIndicator';
+import DragHandle from '../../../../UI/DragHandle';
 
 type Props = {|
+  parentVerticeId: string,
   canRemove: boolean,
   onRemove: () => void,
   verticeX: number,
   verticeY: number,
   onChangeVerticeX: (value: number) => void,
   onChangeVerticeY: (value: number) => void,
+  setDragged: () => void,
+  drop: () => void,
 |};
 
-const VerticeRow = (props: Props) => (
-  <ThemeConsumer>
-    {muiTheme => (
-      <TableRow
-        style={{
-          backgroundColor: muiTheme.list.itemsBackgroundColor,
-        }}
-      >
-        <TableRowColumn />
-        <TableRowColumn style={styles.coordinateColumn}>
-          <TextField
-            margin="none"
-            value={props.verticeX}
-            type="number"
-            id="vertice-x"
-            onChange={(e, value) =>
-              props.onChangeVerticeX(parseFloat(value || 0))
-            }
-          />
-        </TableRowColumn>
-        <TableRowColumn style={styles.coordinateColumn}>
-          <TextField
-            margin="none"
-            value={props.verticeY}
-            type="number"
-            id="vertice-y"
-            onChange={(e, value) =>
-              props.onChangeVerticeY(parseFloat(value || 0))
-            }
-          />
-        </TableRowColumn>
-        <TableRowColumn style={styles.toolColumn}>
-          {!!props.onRemove && (
-            <IconButton
-              size="small"
-              onClick={props.onRemove}
-              disabled={!props.canRemove}
-            >
-              <Delete />
-            </IconButton>
-          )}
-        </TableRowColumn>
-      </TableRow>
-    )}
-  </ThemeConsumer>
-);
+const VerticeRow = ({
+  verticeX,
+  verticeY,
+  parentVerticeId,
+  ...props
+}: Props) => {
+  const DragSourceAndDropTarget = React.useMemo(
+    () => makeDragSourceAndDropTarget(`collision-mask-${parentVerticeId}`),
+    [parentVerticeId]
+  );
+  const [verticeXInputValue, setVerticeXInputValue] = React.useState<string>(
+    verticeX.toString()
+  );
+  const [verticeYInputValue, setVerticeYInputValue] = React.useState<string>(
+    verticeY.toString()
+  );
+  React.useEffect(
+    () => {
+      setVerticeXInputValue(verticeX.toString());
+    },
+    [verticeX]
+  );
+  React.useEffect(
+    () => {
+      setVerticeYInputValue(verticeY.toString());
+    },
+    [verticeY]
+  );
+  return (
+    <ThemeConsumer>
+      {muiTheme => (
+        <DragSourceAndDropTarget
+          beginDrag={() => {
+            props.setDragged();
+            return {};
+          }}
+          canDrag={() => true}
+          canDrop={() => true}
+          drop={() => {
+            props.drop();
+          }}
+        >
+          {({ connectDragSource, connectDropTarget, isOver, canDrop }) =>
+            connectDropTarget(
+              <tr
+                className="MuiTableRow-root"
```

Comment:
Can you explain the reason for this? It's seems a bit risky because there is no guarantee MuiTableRow-root is 1) injected in the CSS already (if we've not used a table in another place, we won't have it) and 2) won't change at the next update.